### PR TITLE
e2e: support `admissionapi.LevelRestricted` in `test/e2e/framework/volume`

### LINKED ISSUE #118184: test/e2e/storage: reduce privileges granted to pods
### What would you like to be added?

Most tests currently run with `f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged`.

Only the namespace created for the CSI driver needs to allow privileged pods. The main test namespace is often just used to test that a pod using a volume can start up, which might work with `admissionapi.LevelRestricted`.

To achieve this, the level of the additional namespace must get decoupled from `f.NamespacePodSecurityEnforceLevel`: 
https://github.com/kubernetes/kubernetes/blob/ce05a4f7fcc234dd3132a4e40e6afdf0b2ebc94e/test/e2e/framework/framework.go#L451-L455

Then [`runVolumeTesterPod`](https://github.com/kubernetes/kubernetes/blob/ce05a4f7fcc234dd3132a4e40e6afdf0b2ebc94e/test/e2e/framework/volume/fixtures.go#L389) must be extended to run pods with a specific level, ideally `admissionapi.LevelRestricted`.

### Why is this needed?

Testing with more privileges than really needed might hide issues that users will run into when trying to harden their cluster setup by granting pods less privileges.

### PR DESCRIPTION
**Pull request by wafuwafu13**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Testing with more privileges than really needed might hide issues that users will run into when trying to harden their cluster setup by granting pods less privileges.

#### Which issue(s) this PR fixes:

Fixes #118184

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/volume/fixtures.go [hunk 1]
```diff
@@ -439,14 +438,6 @@ func runVolumeTesterPod(ctx context.Context, client clientset.Interface, timeout
 	for i, test := range tests {
 		volumeName := fmt.Sprintf("%s-%s-%d", config.Prefix, "volume", i)
 
-		// We need to make the container privileged when SELinux is enabled on the
-		// host,  so the test can write data to a location like /tmp. Also, due to
-		// the Docker bug below, it's not currently possible to map a device with
-		// a privileged container, so we don't go privileged for block volumes.
-		// https://github.com/moby/moby/issues/35991
```
Comment (wafuwafu13) [inline test/e2e/framework/volume/fixtures.go:446]: https://github.com/moby/moby/issues/35991 has already been closed.

File: test/e2e/framework/volume/fixtures.go [hunk 2]
```diff
@@ -547,7 +538,7 @@ func TestVolumeClientSlow(ctx context.Context, f *framework.Framework, config Te
 
 func testVolumeClient(ctx context.Context, f *framework.Framework, config TestConfig, fsGroup *int64, fsType string, tests []Test, slow bool) {
 	timeouts := f.Timeouts
-	clientPod, err := runVolumeTesterPod(ctx, f.ClientSet, timeouts, config, "client", false, fsGroup, tests, slow)
+	clientPod, err := runVolumeTesterPod(ctx, f.ClientSet, timeouts, config, "client", f.NamespacePodSecurityLevel, fsGroup, tests, slow)
```
Comment (pohly) [inline test/e2e/framework/volume/fixtures.go:541]: > Fixes https://github.com/kubernetes/kubernetes/issues/118184

Is that true? That issue is about testing with less privileged pods, but I don't see that in this PR. Many E2E volume tests set `f.NamespacePodSecurityLevel` to `admissionapi.LevelPrivileged`, so there isn't really much of a difference.

**Pull request closed without merging**